### PR TITLE
#1116 fix resign on delay for unfunded projects

### PR DESCRIPTION
**Pull request by krzyk**
#1116 
- added test for validating that neither DEV nor REV jobs are resigned for unfunded projects
- fixed problem in resign_on_delay.groovy

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1474](https://github.com/zerocracy/farm/issues/1474) is now in scope, role is `REV`

File: src/test/resources/com/zerocracy/bundles/dont_resign_on_delay_unfunded_project/_before.groovy [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import com.zerocracy.Project
+import com.zerocracy.entry.ExtGithub
+import com.zerocracy.pm.cost.Ledger
+import org.hamcrest.MatcherAssert
+import org.hamcrest.Matchers
+
+def exec(Project project, XML xml) {
+  Farm farm = binding.variables.farm
+  Github github = new ExtGithub(farm).value()
+  Repo repo = github.repos().create(new Repos.RepoCreate('test', false))
```
Comment (amihaiemil) [inline src/test/resources/com/zerocracy/bundles/dont_resign_on_delay_unfunded_project/_before.groovy]: @krzyk Can we inline ``github``?

File: src/test/resources/com/zerocracy/bundles/dont_resign_on_delay_unfunded_project/_after.groovy [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import org.hamcrest.Matchers
+
+def exec(Project project, XML xml) {
+  MatcherAssert.assertThat(
+    'Issue was resigned',
+    new Orders(project).bootstrap().jobs('lazydev'),
+    Matchers.contains('gh:test/test#1')
+  )
+  MatcherAssert.assertThat(
+    'PR was resigned',
```
Comment (amihaiemil) [inline src/test/resources/com/zerocracy/bundles/dont_resign_on_delay_unfunded_project/_after.groovy]: @krzyk Can you give more detalied assertion-error messages? Here: ``CR should not have been resigned`` and up ``DEV should not have been resigned``

File: src/test/resources/com/zerocracy/bundles/dont_resign_on_delay_unfunded_project/orders.xml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+SOFTWARE.
+-->
+<orders xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://datum.zerocracy.com/0.61.5/xsd/pm/in/orders.xsd" version="0.1" updated="2017-08-18T13:18:00.000Z">
+  <order job="gh:test/test#1">
+    <created>2017-08-18T13:18:00.000Z</created>
+    <performer>lazydev</performer>
+    <reason>some reason</reason>
+  </order>
+  <order job="gh:test/test#2">
+    <created>2017-08-18T13:18:00.000Z</created>
```
Comment (amihaiemil) [inline src/test/resources/com/zerocracy/bundles/dont_resign_on_delay_unfunded_project/orders.xml]: @krzyk Can we use dates closer to the present? Same reasoning as in an older PR: if we decide at some point to not take into account or remove old orders/claims, these tests won't work anymore, I think.

File: src/main/groovy/com/zerocracy/stk/pm/in/orders/resign_on_delay.groovy [hunk 4]
```diff
@@ -16,15 +16,15 @@ import com.zerocracy.pm.in.Orders
...
 import com.zerocracy.pm.staff.Roles
 import com.zerocracy.pmo.Pmo
-import org.cactoos.iterable.Filtered
-import org.cactoos.iterable.Limited
 import java.time.ZoneOffset
 import java.time.ZonedDateTime
+import org.cactoos.iterable.Filtered
+import org.cactoos.iterable.Limited
 
 def exec(Project project, XML xml) {
```
Comment (amihaiemil) [inline src/main/groovy/com/zerocracy/stk/pm/in/orders/resign_on_delay.groovy]: @krzyk While we're still here, can you add a Javadoc on this stakeholder, explaining what it does and it what situations?

**Comment by krzyk** [comment]
@amihaiemil code updated, take a look

**Comment by amihaiemil** [comment]
@krzyk thanks

**Comment by amihaiemil** [comment]
@rultor good to merge

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by amihaiemil** [comment]
@g4s8 can you pls take this job out of scope? 0crat thinks it is still assigned to me

**Comment by g4s8** [comment]
@0crat out

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/zerocracy/farm/issues/1474#issuecomment-408685953))

@g4s8 Job [#1474](https://github.com/zerocracy/farm/issues/1474) is not in the agenda of @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), can't retrieve data and time of add

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/zerocracy/farm/issues/1474#issuecomment-408685953))

@g4s8 Job [#1474](https://github.com/zerocracy/farm/issues/1474) is not in the agenda of @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), can't inspect

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/zerocracy/farm/issues/1474#issuecomment-408685953))

@g4s8 The job [#1474](https://github.com/zerocracy/farm/issues/1474) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@g4s8 this issue is still in my agenda as well

**Comment by ypshenychka** [comment]
@g4s8 ping

**Comment by ypshenychka** [comment]
@g4s8 this job is still in my agenda

**Comment by g4s8** [comment]
@ypshenychka did you receive the payment?

**Comment by g4s8** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/zerocracy/farm/issues/1474#issuecomment-412849925))

@g4s8 This is what I know about this job in [C3NDPUA8L](https://www.0crat.com/p/C3NDPUA8L), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#1474](https://github.com/zerocracy/farm/issues/1474) is not in scope
  * The job is waiting quality review verdict by @ypshenychka[/z](https://www.0crat.com/u/ypshenychka) for 16days
  * @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) will receive 15 minutes after quality review
  * Job [footprint](https://www.0crat.com/footprint/C3NDPUA8L?q=%7Bjob%3A%27gh%3Azerocracy%2Ffarm%231474%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by ypshenychka** [comment]
@g4s8 yes, I was paid for this one

**Comment by g4s8** [comment]
@0crat quality bad

**Comment by 0crat** [comment]
Quality is low, no payment, see [§31](http://www.zerocracy.com/policy.html#31)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/zerocracy/farm/issues/1474#issuecomment-414108886))

@g4s8 This is what I know about this job in [C3NDPUA8L](https://www.0crat.com/p/C3NDPUA8L), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#1474](https://github.com/zerocracy/farm/issues/1474) is not in scope
  * Job [footprint](https://www.0crat.com/footprint/C3NDPUA8L?q=%7Bjob%3A%27gh%3Azerocracy%2Ffarm%231474%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Pull request merged**